piksi_tools/fileio.py

```diff
@@ -142,13 +165,13 @@ def _window_available(self):
         return len(self.write_pool) != 0
 
     def __enter__(self):
-        self.link.add_callback(self._cb, self.msg_type)
+        self.link.add_callback(self._write_cb, self.msg_type)
         return self
 
     def __exit__(self, type, value, traceback):
-        self.link.remove_callback(self._cb, self.msg_type)
+        self.link.remove_callback(self._write_cb, self.msg_type)
 
-    def _cb(self, msg, **metadata):
+    def _write_cb(self, msg, **metadata):
         for pending_write in self.pending[:]:
             if msg.sequence == pending_write.message.sequence:
```

Comment:
do we need a try catch around this comparison?

Can we guarantee that pending_write.message.sequence is a float and is not None at this point?

Reply:
I removed the clear of `message` field from `invalidate` so that this comparison would still work and the PendingWrite record only enters `self.pending` after passing through the `track` method which guarantees that the `message` field is set.